src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -312,9 +312,21 @@ public override IEnumerable<string> CopyFiles(
                     packageFileName = packageFileName.Substring(1);
                 }
 
-                // ZipArchive always has forward slashes in them. By replacing them with DirectorySeparatorChar;
-                // in windows, we get the windows-style path
-                var normalizedPath = Uri.UnescapeDataString(packageFileName.Replace('/', Path.DirectorySeparatorChar));
+                // While NuGet creates packages with forward slashes for directory separators, tools created by other people
+                // often do not normalize to forward slash when creating the zip on Windows. But Linux and Mac do not use
+                // back slashes as directory separators. In fact, back slash is a valid character in a file entry name.
+                // Therefore, normalize the path to ensure that packages created by unofficial tools are not broken
+                // on these platforms.
+                // Windows allows forward slashes as an alternate directory separator, so no normalization is needed there.
+                string normalizedPath;
+                if (RuntimeEnvironmentHelper.IsWindows)
+                {
+                    normalizedPath = Uri.UnescapeDataString(packageFileName);
```

Comment:
Suggestion: maybe we should still replace '/' with Path.DirectorySeperatorChar for windows?  That way we get the platform directory separator. However, it is of course highly likely it will always be '/' for windows.

Reply:
we've made a lot of performance gains by removing unnecessary memory allocations and therefore garbage collections. I'm hoping we could avoid the string allocations. Since this happens for every file in the package, and packages always have multiple files, it's a lot of allocations. It's less impactful on developer machines where a package is extracted into the global packages folder once, but it's very high impact on CI, where packages are typically downloaded and extracted on every build.